src/Raven.Server/Utils/CpuUsageExtensionPoint.cs

```diff
@@ -0,0 +1,284 @@
+        private readonly NotificationCenter.NotificationCenter _notificationCenter;
+        private readonly ProcessStartInfo _startInfo;
+        private readonly TimeSpan _timeout = TimeSpan.FromSeconds(10);
+
+        private Process _process;
+        private bool _didRestart;
+
+        private ExtensionPointRawData _data = new ExtensionPointRawData();
+        public ExtensionPointRawData BadData = new ExtensionPointRawData
+        {
+            ActiveCores = -1,
+            MachineCpuUsage = -1
+        };
+
+        public ExtensionPointRawData Data
+        {
+            get
+            {
+                if (IsDisposed && _logger.IsOperationsEnabled)
+                {
+                    _logger.Operations($"Try to get data from {nameof(CpuUsageExtensionPoint)} which was disposed");
+                }
+                return _data;
+            }
+        }
+
+        public bool IsDisposed { get; private set; }
+
+
+        public CpuUsageExtensionPoint(
+            JsonContextPool contextPool,
+            string exec,
+            string args,
+            Logger logger,
+            NotificationCenter.NotificationCenter notificationCenter)
+        {
+            _contextPool = contextPool;
+            _logger = logger;
+            _notificationCenter = notificationCenter;
+            _startInfo = new ProcessStartInfo
+            {
+                FileName = exec,
+                Arguments = args,
+                UseShellExecute = false,
+                RedirectStandardOutput = true,
+                RedirectStandardError = true,
+                CreateNoWindow = true
+            };
+        }
+
+        public void Start()
+        {
+            _process = new Process
+            {
+                StartInfo = _startInfo,
+                EnableRaisingEvents = true
+            };
+
+            try
+            {
+                _process.Start();
+            }
+            catch (Exception e)
+            {
+                NotifyWarning("Could not start cpu usage extension point process", e);
+
+                Dispose();
+
+                return;
+            }
+
+            var _ = ReadProcess(); // explicitly starting async task without waiting for it
+            _ = ReadErrors();
+        }
+
+        private async Task ReadErrors()
+        {
+            try
+            {
+                var errors = await _process.StandardError.ReadLineAsync();
+                if (errors == null)
+                {
+                    return;
+                }
+                try
+                {
+                    if (_process.HasExited == false)
+                    {
+                        _process.Kill();
+                    }
+                }
+                catch (Exception)
+                {
+                    // When the process terminating, killed or exited
+                }
+                errors = errors + await _process.StandardError.ReadToEndAsync();
```

Comment:
As I understand it the `ReadToEndAsync` would not return until the stream is closed.
So if the process send error without exiting it will continue to wait for the continuation